fastcrypto/src/hash.rs

```diff
@@ -2,94 +2,109 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use base64ct::{Base64, Encoding};
-use digest::OutputSizeUser;
-use generic_array::{ArrayLength, GenericArray};
 use serde::{Deserialize, Serialize};
 use std::fmt;
 
-/// Represents a hash digest of `DigestLength` bytes.
-#[derive(Hash, PartialEq, Eq, Clone, Deserialize, Serialize, Ord, PartialOrd)]
-pub struct Digest<DigestLength: ArrayLength<u8> + 'static>(pub GenericArray<u8, DigestLength>);
+/// The length of a digest.
+pub const DIGEST_LEN: usize = 32;
```

Comment:
Is there a reason to PIN the length? What if in the future we support 384 or 512?

Reply:
It was possible in the previous change of the hash functions, but sui depended quite a bit on the length being constant. I'll gladly give it a go making the length variable and then put the constant in sui, but I think we should consider keeping it constant to make the merge into sui easier, and then add an issue about making the length variable.

Reply:
Actually it wasn't too bad, so I've reintroduced the variable digest length but this time as a constant instead of a typenum. It adds a bit of redundancy a few places, because we need the size as both types when calling the underlying crates, but I managed to keep it inside fastcrypto such that only the constant is exposed.